### PR TITLE
Remove dependency of drake/core/** on other subdirectories.

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
This removes dependency of `drake/core/**` on other subdirectories.  The `core` subdirectory is the first one that the `drake/CMakeLists.txt` visits, so must not rely on files from other, later siblings.

### CODE REVIEW

File: drake/core/test/testVector.cpp [hunk 1]
```diff
@@ -1,9 +1,18 @@
 
-#include "drake/examples/Pendulum/Pendulum.h"  // to get some types
-#include "drake/util/testUtil.h"
+#include "pendulum.h"
```
Comment (sammy-tri) [inline drake/core/test/testVector.cpp]: FWIW, I got dinged in #1962 for not including the full path to a file in the current directory...  The google style guide seems to point in that direction with the dir/foo2.h example in https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

**Comment by jwnimmer-tri** [comment]
Now we wait for all of the builders to vote.

File: drake/core/test/testVector.cpp (hunk 1 above)
Comment (jwnimmer-tri) [inline drake/core/test/testVector.cpp]: Thanks.  Fixed.

**Comment by jwnimmer-tri** [comment]
Need to do a merging and cherry-picking to get this working again, after merge conflicts.  Closing for now.

**Pull request closed without merging**